src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -937,15 +941,24 @@ private bool SynchronizeBatch()
         {
             // Get a batch of blocks.
             int batchSize = 0;
-            HashHeightPair tipToChase = this.TipToChase();
-            Dictionary<uint256, ChainedHeader> chainedHeadersSnapshot = new Dictionary<uint256, ChainedHeader>();
+
+            HashHeightPair federationWalletTip = this.federationWalletManager.LastBlockSyncedHashHeight();
+
+            if (this.chainIndexer.GetHeader(federationWalletTip.Hash) == null)
+            {
+                // Indicates that the federation wallet's tip is not on chain.
+                this.logger.LogTrace("(-)[FEDERATION_WALLET_TIP_NOT_ON CHAIN]:false");
+                return false;
```

Comment:
This is going to lock the node forever in a while loop. The FWM is locked, so it can't progress. The loop in `Synchronize()` is going to hit this over and over until cancellationrequested, aka never

Reply:
I think that this part is correct as if exit here with false, the `while` which contains `SynchronizeBatch` will just call it again and hopefully the federation wallet tip has been updated with a tip that is now on chain. I can't see a scenario where it will never exit?